pkg/gitprovider/gitness_test.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+package gitprovider
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type GitNessGitProviderTestSuite struct {
+	gitProvider *GitNessGitProvider
+	suite.Suite
+}
+
+func NewGitNessGitProviderTestSuite() *GitNessGitProviderTestSuite {
+	baseApiUrl := "http://localhost:3000/api/v1/"
```

Comment:
Please remove tests that require the token to be set. You'll notice that the other providers only have static context parsing as part of their tests.
For this to work, we would need to have an installation of gitness available to our test pipelines.